app/Composers/Modules/ComponentsComposer.php

```diff
@@ -33,9 +34,21 @@ class ComponentsComposer
     public function compose(View $view)
     {
         // Component & Component Group lists.
-        $usedComponentGroups = Component::enabled()->where('group_id', '>', 0)->groupBy('group_id')->pluck('group_id');
-        $componentGroups = ComponentGroup::whereIn('id', $usedComponentGroups)->orderBy('order')->get();
-        $ungroupedComponents = Component::enabled()->where('group_id', 0)->orderBy('order')->orderBy('created_at')->get();
+        $usedComponentGroups = Component::enabled()->where('group_id', '>', 0)
+            ->groupBy('group_id')
+            ->pluck('group_id');
+        $componentGroupsBuilder = ComponentGroup::visible();
+        if (app(Guard::class)->check()) {
+            $componentGroupsBuilder = ComponentGroup::query();
+        }
```

Comment:
I did something a bit different @jbrooksuk and @ConnorVG.

Let me know if it's OK with you.